### PR TITLE
Change Span to refer to SpanStack, rather than TracingContext.

### PR DESCRIPTION
**Pull request by jmjoy**
In order to improve the performance of `TracingContext`, reduce the usage of Arc and Lock.

### CODE REVIEW

File: src/context/trace_context.rs [hunk 1]
```diff
@@ -30,39 +30,86 @@ use crate::{
...
 #[must_use = "call `create_entry_span` after `TracingContext` created."]
 pub struct TracingContext {
-    inner: Arc<Inner>,
+    trace_id: String,
+    trace_segment_id: String,
+    service: String,
+    service_instance: String,
+    next_span_id: i32,
+    span_stack: Arc<SpanStack>,
+    segment_link: Option<PropagationContext>,
```
Comment (wu-sheng) [inline src/context/trace_context.rs:89]: Why is a segment link? Segment reference should be inside a span, and it should be multiple references.

Comment (jmjoy) [reply src/context/trace_context.rs:89]: This field existed before I was involved in this project. 😂

Comment (wu-sheng) [reply src/context/trace_context.rs:89]: Let's check why it exists. and try to remove it?

Comment (jmjoy) [reply src/context/trace_context.rs:89]: Ok, actually I also think this field is not very right.

**Comment by jmjoy** [comment]
Remove TracingContext `segment_link` field, and add TracingContext `create_entry_span_with_propagation` method.

**Comment by wu-sheng** [comment]
> TracingContext create_entry_span_with_propagation method.

This is good. 

A tip, trace ID  could be generated as a new trace, and then do `Inject` later to override the pre generated, when injecting context happens after span creation. This is not necessary, but we do this in Java.

**Review by wu-sheng — APPROVED** [review]
LGTM. Feel free to merge, notice, please clear the commits in the merge message.

**Comment by jmjoy** [comment]
> > TracingContext create_entry_span_with_propagation method.
> 
> This is good.
> 
> A tip, trace ID could be generated as a new trace, and then do `Inject` later to override the pre generated, when injecting context happens after span creation. This is not necessary, but we do this in Java.

OK, I'll take a look later.

**Pull request merged**